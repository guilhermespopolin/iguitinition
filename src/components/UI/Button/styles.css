.button {
  display: inline-block;
  padding: 0.5rem;
  color: var(--white);
  background-color: var(--primary);
  border: none;
  outline: none;
  border-radius: 3px;
}

.button:hover {
  cursor: pointer;
  background-color: var(--primaryLight);
}