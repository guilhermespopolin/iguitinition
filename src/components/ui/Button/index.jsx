import styled from 'styled-components'

const Button = styled.button`
  display: inline-block;
  padding: 0.5rem;
  color: ${({ theme }) => theme.colors.white};
  background-color: ${({ theme }) => theme.colors.primary};
  border: none;
  outline: none;
  border-radius: 3px;

  &:hover {
    cursor: pointer;
    background-color: ${({ theme }) => theme.colors.primaryLight};
  }
`

export default Button
