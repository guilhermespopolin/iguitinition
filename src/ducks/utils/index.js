import { keyBy } from 'lodash'

export function normalizePayload(payload) {
  return keyBy(payload, 'id')
}

export default {
  normalizePayload,
}
