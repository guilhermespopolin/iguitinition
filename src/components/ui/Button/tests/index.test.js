import React from 'react'
import { render } from '__tests__/utils'

import Button from '..'

describe('<Button />', () => {
  it('should render properly', () => {
    const { getByText } = render(<Button>Click Me</Button>)
    const buttonNode = getByText('Click Me')

    expect(buttonNode).toMatchSnapshot()
  })
})

