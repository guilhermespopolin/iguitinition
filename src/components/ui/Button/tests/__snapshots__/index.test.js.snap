// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render properly 1`] = `
.c0 {
  display: inline-block;
  padding: 0.5rem;
  color: #fefefe;
  background-color: #0a30ad;
  border: none;
  outline: none;
  border-radius: 3px;
}

.c0:hover {
  cursor: pointer;
  background-color: #1542d4;
}

<button
  class="c0"
>
  Click Me
</button>
`;
